Docs(Deploy): updated deploy / administration / exports section (#60)
dgraph-io/dgraph-docs: content/deploy/dgraph-administration.md

The user must also ensure that there is sufficient space on disk to store the
export.

> ### Configuring Dgraph Alpha server

Reviewer: Suggested tweak to the heading: "Configuring Dgraph Alpha server nodes"

Each Dgraph Alpha leader for a group writes output as a gzipped file to the export
directory specified via the `--export` flag (defaults to a directory called `"export"`). If any of the groups fail, the
entire export process is considered failed and an error is returned.

> As an example of configuring `export`, you can run this:

Reviewer: Suggested edit:
As an example of configuring `export`, you can run this: -> For example, you could run the following commands to configure data exports:

Author: @aaroncarey I wanted to mention that exports can be configured in 3 ways (env var, command line argument, and configuration file, and see <link> for more info), but not sure how to say this and not break the current flow.

```bash
docker run --detach --rm --name dgraph-standalone \
  --publish 8080:8080 \
  --publish 8000:8000 \
  --volume ~/exports:/dgraph/myexports \
  --env "DGRAPH_ALPHA_EXPORT=/dgraph/myexports" \
  dgraph/standalone:{{< version >}}
```

### Export data format

> The data is exported in RDF format by default.  You can explicitly set the output format with the `format` field. For example:

Reviewer: Suggested edit:

The data is exported in RDF format by default. -> By default, Dgraph exports data in RDF format.

(avoids passive voice)

}
```

> A different output format may be specified with the `format` field. For example:

Reviewer: A different output format may be specified with the `format` field. -> You can specify a different output format using the `format` field.

(avoids passive voice)

}
```

Currently, `rdf` and `json` are the only formats supported.

### Exporting to NFS or a file path

{{% notice "note" %}}
This feature was introduced in [v20.11.0](https://github.com/dgraph-io/dgraph/releases/tag/v20.11.0).
{{% /notice %}}

> You can override the default folder path by adding `destination` input field to the directory where you want to export data in the GraphQL mutation request as follows:

Reviewer: adding `destination` -> adding the `destination`

request as follows: -> request, as follows:

```graphql
mutation {
  export(input: {
    format: "rdf"
    destination: "<absolute-path-to-your-export-dir>"
  }) {
    response {
      message
      code
    }
  }
}
```

### Exporting to an object store

{{% notice "note" %}}
This feature was introduced in [v20.11.0](https://github.com/dgraph-io/dgraph/releases/tag/v20.11.0).
{{% /notice %}}

You can export to an S3 or MinIO object store by specifying a URL in the `destination` input field.

#### Exporting to S3

```graphql
mutation {
  export(input: {
    destination: "s3://s3.<region>.amazonaws.com/<bucket-name>"
    accessKey: "<aws-access-key-id>"
    secretKey: "<aws-secret-access-key>"
  }) {
    response {
      message
      code
    }
  }
}
```

{{% notice "note" %}}

> Dgraph URL used for s3 is different than AWS CLI tools with the `aws s3` command, which uses a shortened format: `s3://<bucket-name>`.

Reviewer 2: ```suggestion
Dgraph URL used for S3 is different than AWS CLI tools with the `aws s3` command, which uses a shortened format: `s3://<bucket-name>`.
```

{{% /notice %}}

#### Exporting to MinIO

```graphql
mutation {
  export(input: {
    destination: "minio://<address>:9000/<bucket-name>"
    accessKey: "<minio-access-key>"
    secretKey: "<minio-secret-key>"
  }) {
    response {
      message
      code
    }
  }
}
```

#### Exporting to a MinIO Gateway

> MinIO can be used as a gateway to other object stores, such as [Azure Blob Storage](https://azure.microsoft.com/services/storage/blobs/) or [Google Cloud Storage](https://cloud.google.com/storage).  You can use the above MinIO GraphQL mutation with MinIO configured as a gateway.

Reviewer: MinIO can be used -> You can use MinIO

##### Azure Blob Storage

> You can use [Azure Blob Storage](https://azure.microsoft.com/services/storage/blobs/) through the [MinIO Azure Gateway](https://docs.min.io/docs/minio-gateway-for-azure.html).  You need to configure a [storage account](https://docs.microsoft.com/azure/storage/common/storage-account-overview) and a Blob [container](https://docs.microsoft.com/azure/storage/blobs/storage-blobs-introduction#containers) to organize the blobs.  The blob container name synonymous to bucket name when specifying the destination.  

Reviewer: "The blob container name synonymous to bucket name when specifying the destination. "

Does this mean the following?:

"The name of the blob container should be the same as the name of the storage bucket you use when specifying the destination."

Reviewer: ... I made this edit in my "patch-in"; please correct if my reading was off-base.

Author: In this case, you create a "blob container name", which is the bucket name in MinIO.  I wanted to say these are the same thing, to avoid confusion in terminology between two different technologies.  If users don't know what the "bucket name" should, then use the "blob container name"

   ```

#### Disabling HTTPS for exports to S3 and Minio

By default, Dgraph assumes the destination bucket is using HTTPS. If that is not the case, the export will fail. To export to a bucket using HTTP (insecure), set the query parameter `secure=false` with the destination endpoint in the destination field:

```graphql
mutation {
  export(input: {
    destination: "minio://<address>:9000/<bucket-name>?secure=false"
    accessKey: "<minio-access-key>"
    secretKey: "<minio-secret-key>"
  }) {
    response {
      message
      code
    }
  }
}
```

#### Using anonymous credentials

If exporting to S3 or MinIO where credentials are not required, you can set `anonymous` to true.

```graphql
mutation {
  export(input: {
    destination: "s3://s3.<region>.amazonaws.com/<bucket-name>"
    anonymous: true
  }) {
    response {
      message
      code
    }
  }
}
```

### Encrypting exports

Export is available wherever an Alpha is running. To encrypt an export, the Alpha must be configured with the `encryption-key-file`.

{{% notice "note" %}}
The `encryption-key-file` was used for `encryption-at-rest` and will now also be used for encrypted backups and exports.
{{% /notice %}}

> ### Using curl to trigger an export

Reviewer 2: ```suggestion
### Using `curl` to trigger an export
```

Author: @bucanero Doesn't that look weird for the heading?

> This is an example in how you can use `curl` to tigger an export.

Reviewer 2: ```suggestion
This is an example of how you can use `curl` to trigger an export.
```

  1. Create GraphQL file for the desired mutation:
     ```bash
     cat <<-EOF > export.graphql
     mutation {
       export(input: {
         destination: "s3://s3.<region>.amazonaws.com/<bucket-name>"
         accessKey: "<aws-access-key-id>"
         secretKey: "<aws-secret-access-key>"
       }) {
         response {
           message
           code
         }
       }
     }
     EOF
     ```

>   2. Trigger an export with curl

Reviewer 2: ```suggestion
  2. Trigger an export with `curl`
```
